apiserver/worker/compiler.py

```diff
@@ -302,6 +302,10 @@ def compile(self, bot_dir, globs, errors, timelimit):
     "D": [
         ["dmd", "-O", "-inline", "-release", "-noboundscheck", "-of" + BOT],
     ],
+    "Elixir": [
+        ["mix", "deps.get"]
+        ["mix", "escript.build"],
+    ],
```

Comment:
Hmm, this is a little more complicated now. One, it requires interactive input:
```
$ mix deps.get --force
Could not find Hex, which is needed to build dependency :logger_file_backend
Shall I install Hex? (if running non-interactively, use "mix local.hex --force") [Yn]
```

If I do that, everything else seems fine. Should this just be added to the list of commands?

Does `mix` install dependencies locally (ala `node_modules` in NPM) or does it install them in some global or user directory? If the latter, that would cause problems (they won't get included in the zip).

Also, this is rather confusing, but the compiler script won't actually run these commands automatically (yeah, I know…) In the `Language` definition for Elixir below, in the same list as the `ChmodCompiler` command, you'll have to specify that you want to invoke these commands, like D does:

```
(["*.d"], ExternalCompiler(comp_args["D"][0]))
```

The first part of the tuple is a list of globs to pass to the compiler specified; the second part is the actual command to execute (which refers back to the `comp_args` here). So the final list (below) would look something like 

```
[
    ([], ErrorFilterCompiler(com_args["Elixir"][0])),
    ([], ErrorFilterCompiler(com_args["Elixir"][1])),
    (["*.ex"], ChmodCompiler("Elixir")),
]
```

If you want, I can tweak it and submit a PR to your branch.

Finally, there's still a mismatch in terms of the file it looks for (as the 'bot' file) and the file it runs. Probably we should have

```
Language("Elixir", BOT, "mix.exs",
```
which specifies that we detect that the bot is Elixir by looking for a file `mix.exs`, which after we run the compiler commands, produces a file called `MyBot`.